src/applications/personalization/profile-2/components/personal-information/PersonalInformation.jsx

```diff
@@ -1,16 +1,36 @@
 import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Prompt } from 'react-router-dom';
+import { connect } from 'react-redux';
 
 import { focusElement } from 'platform/utilities/ui';
 
 import PersonalInformationContent from './PersonalInformationContent';
 
-const PersonalInformation = () => {
+const PersonalInformation = ({ hasUnsavedEdits }) => {
   useEffect(() => {
     focusElement('[data-focus-target]');
   }, []);
 
+  useEffect(
+    () => {
+      // Show alert when navigating away
+      if (hasUnsavedEdits) {
+        window.onbeforeunload = () => true;
+        return;
```

Comment:
Just curious if you have a strong opinion about using a `return` here vs. adding an `else` clause. I would have done the latter. It doesn't really matter but I was wondering about your thought process.